packages/agents/lighthouse/src/tasks/processFromRoot/operations/processFromRoot.ts

```diff
@@ -26,122 +27,186 @@ import {
   getProcessFromMetisRootArgs,
   getProcessFromModeRootArgs,
   getProcessFromScrollRootArgs,
-  getProcessFromXlayerRootArgs,
+  getProcessFromXlayerRootWriteTransaction,
 } from "../helpers";
 import { getContext } from "../processFromRoot";
+import { WriteTransaction } from "@connext/nxtp-txservice";
+import { constants } from "ethers";
 
 export type ProcessConfig = {
-  getArgs: (params: GetProcessArgsParams) => Promise<any[]>;
-  hubConnectorPrefix: string;
-  processorFunctionName: string;
-  contractAddress?: string;
-  contractAbi?: string[];
+  getWriteTransaction: (params: GetProcessArgsParams) => Promise<WriteTransaction | undefined>;
```

Comment:
Like this pattern.
Concerned that this might cause regression on other (non- XLayer) changes, since this touches processFromRoot on all chains.

Reply:
possible, but that's why I grouped the write tx generation in a single function and kept the original `getArgs` utils